CMakeLists.txt

```diff
@@ -0,0 +1,136 @@
+				DEPENDS kafka_to_avro mcskafka mxs_adapter pdi-plugin
+		)
+
+		add_custom_command(
+			TARGET package
+			POST_BUILD
+			COMMAND ${CMAKE_COMMAND} -E rename "*-avro_kafka_adapter.rpm" "${CPACK_RPM_AVRO_KAFKA_ADAPTER_FILE_NAME}"
+			COMMAND ${CMAKE_COMMAND} -E rename "*-mcskafka.rpm" "${CPACK_RPM_MCSKAFKA_FILE_NAME}"
+			COMMAND ${CMAKE_COMMAND} -E rename "*-mxs_adapter.rpm" "${CPACK_RPM_MXS_ADAPTER_FILE_NAME}"
+			WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+			COMMENT "-- Renaming packet file names"
+		)
+	else()
+		include(CPack)
+	endif()
+ENDIF(RPM)
+
+option(DEB "Build a DEB" OFF)
+IF(DEB)
+	SET(CPACK_GENERATOR "DEB")
+	SET(CPACK_DEBIAN_PACKAGE_VERSION ${VERSION})
+	SET(CPACK_DEBIAN_PACKAGE_RELEASE ${RELEASE})
+	SET(CPACK_DEBIAN_PACKAGE_NAME "mariadb-columnstore-data-adapters")
+	SET(CPACK_PACKAGE_CONTACT "MariaDB Corporation")
+	SET(ENGINE_ARCH "x86_64")
+	SET(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+	SET(CPACK_PACKAGE_FILE_NAME "${CPACK_DEBIAN_PACKAGE_NAME}-${CPACK_DEBIAN_PACKAGE_VERSION}-${CPACK_DEBIAN_PACKAGE_RELEASE}-${ENGINE_ARCH}-${DEB}")
+	SET(CPACK_DEB_COMPONENT_INSTALL ON)
+	SET(CPACK_COMPONENTS_ALL avro_kafka_adapter mcskafka mxs_adapter)
+
+	SET(CPACK_DEBIAN_AVRO_KAFKA_ADAPTER_PACKAGE_NAME "mariadb-columnstore-kafka-avro-adapters")
+	SET(CPACK_DEBIAN_AVRO_KAFKA_ADAPTER_FILE_NAME "${CPACK_DEBIAN_AVRO_KAFKA_ADAPTER_PACKAGE_NAME}-${CPACK_DEBIAN_PACKAGE_VERSION}-${CPACK_DEBIAN_PACKAGE_RELEASE}-${ENGINE_ARCH}-${DEB}.deb")
+	
+	SET(CPACK_DEBIAN_MCSKAFKA_PACKAGE_NAME "mariadb-columnstore-kafka-adapters")
+	SET(CPACK_DEBIAN_MCSKAFKA_FILE_NAME "${CPACK_DEBIAN_MCSKAFKA_PACKAGE_NAME}-${CPACK_DEBIAN_PACKAGE_VERSION}-${CPACK_DEBIAN_PACKAGE_RELEASE}-${ENGINE_ARCH}-${DEB}.deb")
+
+	SET(CPACK_DEBIAN_MXS_ADAPTER_PACKAGE_NAME "mariadb-columnstore-maxscale-cdc-adapters")
+	SET(CPACK_DEBIAN_MXS_ADAPTER_FILE_NAME "${CPACK_DEBIAN_MXS_ADAPTER_PACKAGE_NAME}-${CPACK_DEBIAN_PACKAGE_VERSION}-${CPACK_DEBIAN_PACKAGE_RELEASE}-${ENGINE_ARCH}-${DEB}.deb")
+
+	# rename packets manually if CMAKE < 3.6.3
+	if(${CMAKE_VERSION} VERSION_LESS "3.6.3") 
+		SET( CPACK_OUTPUT_CONFIG_FILE "${CMAKE_BINARY_DIR}/DEBPackageConfig.cmake" )
+		include(CPack)
+
+		ADD_CUSTOM_TARGET ( package
+			COMMAND "${CMAKE_CPACK_COMMAND}" 
+				"--config" "${CMAKE_BINARY_DIR}/DEBPackageConfig.cmake"
+				COMMENT "-- Running CPack..."
+				DEPENDS kafka_to_avro mcskafka mxs_adapter pdi-plugin
+		)
+
+		add_custom_command(
+			TARGET package
+			POST_BUILD
+			COMMAND ${CMAKE_COMMAND} -E rename "*-avro_kafka_adapter.deb" "${CPACK_DEBIAN_AVRO_KAFKA_ADAPTER_FILE_NAME}"
+			COMMAND ${CMAKE_COMMAND} -E rename "*-mcskafka.deb" "${CPACK_DEBIAN_MCSKAFKA_FILE_NAME}"
+			COMMAND ${CMAKE_COMMAND} -E rename "*-mxs_adapter.deb" "${CPACK_DEBIAN_MXS_ADAPTER_FILE_NAME}"
+			WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+			COMMENT "-- Renaming packet file names"
+		)
+	else()
+		include(CPack)
+	endif()
+ENDIF(DEB)
+
+option(TEST_RUNNER "Build the test suite" OFF)
+if (TEST_RUNNER)
+  enable_testing()
+  include(CTest)
```

Comment:
CTest should be included as the last thing in the file to allow subdirectories to define tests.